deployment/terraform/strings.go

```diff
@@ -41,3 +41,97 @@ scrape_configs:
     static_configs:
         - targets: ['localhost:9100', %s]
 `
+
+const nginxConfig = `
+user www-data;
+worker_processes auto;
+worker_rlimit_nofile 65536;
+pid /run/nginx.pid;
+include /etc/nginx/modules-enabled/*.conf;
+
+events {
+	worker_connections 200000;        
+}
+
+http {
+	sendfile on;
+	tcp_nopush on;
+	tcp_nodelay on;
+	keepalive_timeout 65;
+	types_hash_max_size 2048;
+	include /etc/nginx/mime.types;
+	default_type application/octet-stream;
+	ssl_prefer_server_ciphers on;
+	access_log /var/log/nginx/access.log;
+	error_log /var/log/nginx/error.log;
+	gzip on;
+	include /etc/nginx/conf.d/*.conf;
+	include /etc/nginx/sites-enabled/*;
+}
+`
+
+const nginxSiteConfig = `
+upstream backend {
+%s
+        keepalive 32;
+}
+
+proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=mattermost_cache:10m max_size=3g inactive=120m use_temp_path=off;
+
+server {
+    listen 80;
+    server_name _;
+
+   location ~ /api/v[0-9]+/(users/)?websocket$ {
+           proxy_set_header Upgrade $http_upgrade;
+           proxy_set_header Connection "upgrade";
+           client_max_body_size 50M;
+           proxy_set_header Host $http_host;
+           proxy_set_header X-Real-IP $remote_addr;
+           proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+           proxy_set_header X-Forwarded-Proto $scheme;
+           proxy_set_header X-Frame-Options SAMEORIGIN;
+           proxy_buffers 256 16k;
+           proxy_buffer_size 16k;
+           client_body_timeout 60;
+           send_timeout        300;
+           lingering_timeout   5;
+           proxy_connect_timeout   90;
+           proxy_send_timeout      300;
+           proxy_read_timeout      90s;
+           proxy_pass http://backend;
+   }
+
+   location / {
+           client_max_body_size 50M;
+           proxy_set_header Connection "";
+           proxy_set_header Host $http_host;
+           proxy_set_header X-Real-IP $remote_addr;
+           proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+           proxy_set_header X-Forwarded-Proto $scheme;
+           proxy_set_header X-Frame-Options SAMEORIGIN;
+           proxy_buffers 256 16k;
+           proxy_buffer_size 16k;
+           proxy_read_timeout 600s;
+           proxy_cache mattermost_cache;
+           proxy_cache_revalidate on;
+           proxy_cache_min_uses 2;
+           proxy_cache_use_stale timeout;
+           proxy_cache_lock on;
+           proxy_http_version 1.1;
+           proxy_pass http://backend;
+   }
+}
+`
+
+const limitsConfig = `
+* soft nofile 65536
+* hard nofile 65536
+* soft nproc 8192
+* hard nproc 8192
+`
+
+const sysctlConfig = `
+net.ipv4.ip_local_port_range="1024 65000"
```

Comment:
This syntax is actually incorrect and I have been meaning to fix https://docs.mattermost.com/deployment/cluster.html?highlight=ip_local_port_range#configuration-settings. We need to get rid of the double quotes. Otherwise it gives:

```
ubuntu@ip-172-31-17-4:~/mattermost$ sudo sysctl -p
sysctl: setting key "net.ipv4.ip_local_port_range": Invalid argument
net.ipv4.ip_local_port_range = "1024 65000"
net.ipv4.tcp_fin_timeout = 30
net.ipv4.tcp_tw_reuse = 1
```

Here is how the correct version looks on my machine:
```
net.ipv4.ip_local_port_range = 1024 65000
net.ipv4.tcp_fin_timeout = 30
```